Release Drafter crashes with "Cannot read property 'login' of null in releases.js" when a pull request submitter is deleted

Hi all! We use the GitHub app to run Release Drafter in the Jenkins project. We use a [centralized configuration](https://github.com/jenkinsci/.github/blob/master/.github/release-drafter.yml) which runs across 80 repositories ATM. Currently it works for the most of the repositories really well, except the diagnosability issues (I'd guess migration to GitHub Actions or self-hosting would be a right step for our project scale).

Yesterday we hit an issue with https://github.com/jenkinsci/gradle-plugin . The release drafter was enabled there yesterday, and the App was enabled for the repo. Everything looks fine AFAICT, but the changes are not being generated.

It would be great if somebody could take a look at the application logs so that we kno the root cause of the issue

CC @wolfs

This is the output from GitHub Actions when an app has no releases and you run release-drafter. I've redacted the repo name in the output.

```
2019-09-13T13:47:59.7322134Z ##[group]Run toolmantim/release-drafter@v5.2.0
2019-09-13T13:47:59.7400690Z env:
2019-09-13T13:47:59.7401226Z   GITHUB_TOKEN: ***
2019-09-13T13:47:59.7401366Z ##[endgroup]
2019-09-13T13:47:59.7759734Z ##[command]/usr/bin/docker run --name df7dc3b0d3721859e41eea1a5c7972ec92e4a_a6e960 --label 0df7dc --workdir /github/workspace --rm -e GITHUB_TOKEN -e HOME -e GITHUB_REF -e GITHUB_SHA -e GITHUB_REPOSITORY -e GITHUB_ACTOR -e GITHUB_WORKFLOW -e GITHUB_HEAD_REF -e GITHUB_BASE_REF -e GITHUB_EVENT_NAME -e GITHUB_WORKSPACE -e GITHUB_ACTION -e GITHUB_EVENT_PATH -e RUNNER_OS -e RUNNER_TOOL_CACHE -e RUNNER_TEMP -e RUNNER_WORKSPACE -v "/var/run/docker.sock":"/var/run/docker.sock" -v "/home/runner/work/_temp/_github_home":"/github/home" -v "/home/runner/work/_temp/_github_workflow":"/github/workflow" -v "/home/runner/work/***/***":"/github/workspace" 0df7dc:3b0d3721859e41eea1a5c7972ec92e4a
2019-09-13T13:48:01.4825178Z 13:48:01.477Z  INFO probot: creativegs/***: Found 0 releases
2019-09-13T13:48:01.4826208Z 13:48:01.479Z  INFO probot: creativegs/***: No draft release found
2019-09-13T13:48:01.4826586Z 13:48:01.479Z  INFO probot: creativegs/***: No last release found
2019-09-13T13:48:01.4826970Z 13:48:01.479Z  INFO probot: creativegs/***: Fetching all commits for branch master
2019-09-13T13:49:55.2448956Z 13:49:55.243Z ERROR event: Cannot read property 'login' of null (id=a0828448-2b0b-4038-9c95-2316263227a4)
2019-09-13T13:49:55.2449675Z   TypeError: Cannot read property 'login' of null
2019-09-13T13:49:55.2449854Z       at acc.push.categorizedPullRequests.(anonymous function).map.pullRequest (/app/lib/releases.js:153:43)
2019-09-13T13:49:55.2450010Z       at Array.map (<anonymous>)
2019-09-13T13:49:55.2450146Z       at orderedCategories.reduce (/app/lib/releases.js:149:12)
2019-09-13T13:49:55.2450281Z       at Array.reduce (<anonymous>)
2019-09-13T13:49:55.2450411Z       at generateChangeLog (/app/lib/releases.js:136:6)
2019-09-13T13:49:55.2452217Z       at module.exports.generateReleaseInfo (/app/lib/releases.js:179:17)
2019-09-13T13:49:55.2452374Z       at app.on (/app/index.js:69:25)
2019-09-13T13:49:55.2452752Z   --
2019-09-13T13:49:55.2452882Z   event: {
2019-09-13T13:49:55.2452951Z     "event": "push",
2019-09-13T13:49:55.2453258Z     "id": "a0828448-2b0b-4038-9c95-2316263227a4",
2019-09-13T13:49:55.2453562Z     "repository": "creativegs/***"
2019-09-13T13:49:55.2453685Z   }
2019-09-13T13:49:55.6875205Z ##[error]Docker run failed with exit code 1
```

--- a/lib/releases.js
+++ b/lib/releases.js
@@ -62,7 +62,9 @@ const contributorsSentence = ({ commits, pullRequests }) => {
   })
 
   pullRequests.forEach(pullRequest => {
-    contributors.add(`@${pullRequest.author.login}`)
+    if (pullRequest.author) {
+      contributors.add(`@${pullRequest.author.login}`)
+    }
   })
 
   const sortedContributors = Array.from(contributors).sort()

--- a/lib/releases.js
+++ b/lib/releases.js
@@ -150,7 +150,7 @@ const generateChangeLog = (mergedPullRequests, config) => {
             template(config['change-template'], {
               $TITLE: pullRequest.title,
               $NUMBER: pullRequest.number,
-              $AUTHOR: pullRequest.author.login
+              $AUTHOR: pullRequest.author ? pullRequest.author.login : "unknown"
             })
           )
           .join('\n')

--- a/lib/releases.js
+++ b/lib/releases.js
@@ -152,7 +152,7 @@ const generateChangeLog = (mergedPullRequests, config) => {
             template(config['change-template'], {
               $TITLE: pullRequest.title,
               $NUMBER: pullRequest.number,
-              $AUTHOR: pullRequest.author ? pullRequest.author.login : "unknown"
+              $AUTHOR: pullRequest.author ? pullRequest.author.login : "ghost"
             })
           )
           .join('\n')

